.github/workflows/main.yml

```diff
@@ -7,34 +7,40 @@ on:
   pull_request:
     types: [opened, synchronize, reopened]
 
+env:
+  PNPM_VERSION: 9.1.1
+
 jobs:
   build:
-    name: Build, lint, and test on Node ${{ matrix.node }} and ${{ matrix.os }}
-
-    runs-on: ${{ matrix.os }}
+    name: Build, lint, and test on Node ${{ matrix.node }}
+    runs-on: ubuntu-latest
     strategy:
       matrix:
         node: ["20.x"]
-        os: [ubuntu-latest]
 
     steps:
       - name: Checkout repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v3.0.0
+        with:
+          version: ${{ env.PNPM_VERSION }}
```

Comment:
pnpm breaks sometimes with diff versions, even when they are fix/feat updates, so better to write the current version.